### PR TITLE
New search layout

### PR DESCRIPTION
**Pull request by SaraVieira**

### CODE REVIEW

File: components/Search/Item.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+const getHighlight = ({ title, search, description, content }) => {
+  const s = search.toLowerCase()
+
+  if (title.toLowerCase().includes(s)) {
+    return {
+      type: 'title',
+      highlight: title.toLowerCase().indexOf(s),
+    }
+  }
+  console.log(description)
```
Comment (CodyJasonBennett) [inline components/Search/Item.tsx:14]: We'll need to remove this before we deploy. Same with L36.

File: utils/removeMarkdown.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      .replace(/<[^>]*>/g, '')
+      .replace(/^[=\-]{2,}\s*$/g, '')
+      .replace(/\[(.*?)\][\[\(].*?[\]\)]/g, '$1')
+
+      .replace(/^\s{0,3}>\s?/g, '')
+      .replace(/^\s{1,2}\[(.*?)\]: (\S+)( ".*?")?\s*$/g, '')
+      // Remove atx-style headers
+      .replace(/^(\n)?\s{0,}#{1,6}\s+| {0,}(\n)?\s{0,}#{0,} {0,}(\n)?\s{0,}$/gm, '$1$2$3')
+      // Remove emphasis (repeat the line to remove double emphasis)
+      .replaceAll('**', '')
```
Comment (CodyJasonBennett) [inline utils/removeMarkdown.ts:13]: I don't think we can use replaceAll yet. Vercel uses Node 14 and [this was added in v15](https://nodejs.org/en/blog/release/v15.0.0/#v8-8-6-35415).

Comment (CodyJasonBennett) [inline utils/removeMarkdown.ts:13]: I don't think we can use replaceAll yet. Vercel uses Node 14 and [this was added in v15](https://nodejs.org/en/blog/release/v15.0.0/#v8-8-6-35415).

File: utils/mdxUtils.ts [hunk 3]
```diff
@@ -44,13 +44,13 @@ export const getAllDocs = async () => {
...
       const pathname = url.split('/').pop()
-      const folder = url.split('/')[2]
 
       return {
         url,
         title: data.title || pathname.replace(/\-/g, ' '),
+        description: data.description || '',
         nav: data.nav ?? Infinity,
-        content,
+        content: content,
```
Comment (CodyJasonBennett) [inline utils/mdxUtils.ts:53]: We'll need to clean this up to maintain code style.

File: components/Search/Item.tsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  return (
+    <Link href={url}>
+      <a className="block no-underline search-item outline-none">
+        <li className={'px-2 py-1'}>
+          <div className="  p-4 py-5 rounded-md bg-gray-100 hover:bg-gray-800 hover:text-gray-200  flex justify-between items-center transition-all">
+            <span>
+              {multipleLibs ? (
+                <span className="text-xs font-light text-gray-500 block pb-1">{name}</span>
+              ) : null}
```
Comment (CodyJasonBennett) [inline components/Search/Item.tsx:44]: I don't believe we have to use ternaries here. `multipleLibs && <span />` should do.

**Review by CodyJasonBennett — CHANGES_REQUESTED** [review]

File: components/mdx/Demo.tsx [hunk 5]
```diff
@@ -94,50 +101,36 @@ font-weight: bold;
...
-      <DemoContent>
+
+      <div className="relative h-full overflow-hidden">
         <ErrorBoundary>
           {inViewport && data ? (
             <SandpackRunner
               customSetup={{
                 files: data,
-                entry: `/${JSON.parse(data['/package.json']).main}`,
+                entry: `/ ${JSON.parse(data['/package.json']).main}`,
```
Comment (CodyJasonBennett) [inline components/mdx/Demo.tsx:133]: Is this space intentional?

File: utils/mdxUtils.ts (hunk 3 above)
Comment (SaraVieira) [reply utils/mdxUtils.ts]: Fixed!

File: components/Search/Item.tsx (hunk 4 above)
Comment (SaraVieira) [reply components/Search/Item.tsx]: Fixed

Comment (SaraVieira) [reply components/Search/Item.tsx]: Also cleaned up this file a bunch

File: components/mdx/Demo.tsx (hunk 5 above)
Comment (SaraVieira) [reply components/mdx/Demo.tsx:133]: god no

Comment (SaraVieira) [reply components/mdx/Demo.tsx:133]: Fixed

**Pull request merged**